presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileReader.java

```diff
@@ -200,11 +200,17 @@ else if (SEQUENCE_FILE_MAGIC.equals(magic)) {
         syncFirst = input.readLong();
         syncSecond = input.readLong();
 
-        // seek to first row group
+        // seek to first row group, unless the region starts within the file header
         if (offset > input.position()) {
+            // if the specified file region does not contain the start of a sync sequence, this call will close the reader
             seekToFirstRowGroupInRange(offset, length);
         }
-        // if the specified file region does not contain the start of a sync sequence, advance() will close the file on the first call
+        else if (offset != 0) {
+            // The reader that covers the first byte of the file owns all row groups up to the first
+            // sync point.  The region for this reader is completely contained within the header, so
```

Comment:
The tests you suggested found problems with this code.  I pushed a fix to the end of this chain (along with the new test)